From 9c63480ed3aa5c23ef9b6c0ead8faa813650fbe2 Mon Sep 17 00:00:00 2001
From: Jan Petrous <jan.petrous@nxp.com>
Date: Fri, 19 Feb 2021 14:23:18 +0100
Subject: [PATCH 77/84] dts: s32g27a: pfe: add PFE slave driver entry example

The example of PFE Master/Slave entries was added.
In fact that Master can reuse 'ordinary' pfe entry,
only additional property was necessary to add to
signal which HIF will be used for IHC communication.

More info can be found in PFE Linux UM.

Issue: ALB-6478
Signed-off-by: Jan Petrous <jan.petrous@nxp.com>
---
 .../boot/dts/freescale/fsl-s32g274a-evb.dts   |  8 ++++
 .../boot/dts/freescale/fsl-s32g274a-rdb2.dts  | 12 +++++-
 .../boot/dts/freescale/fsl-s32g274a.dtsi      | 41 +++++++++++++++++++
 3 files changed, 59 insertions(+), 2 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/fsl-s32g274a-evb.dts b/arch/arm64/boot/dts/freescale/fsl-s32g274a-evb.dts
index 6914acbeb05d..df56626af36d 100644
--- a/arch/arm64/boot/dts/freescale/fsl-s32g274a-evb.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-s32g274a-evb.dts
@@ -72,6 +72,14 @@
 	};
 };
 
+&pfe_reserved_slave {
+	status = "disabled";
+};
+
+&pfe_slave {
+	status = "disabled";
+};
+
 &pfe {
 	status = "okay";
 	memory-region = <&pfe_reserved>;
diff --git a/arch/arm64/boot/dts/freescale/fsl-s32g274a-rdb2.dts b/arch/arm64/boot/dts/freescale/fsl-s32g274a-rdb2.dts
index c4fc76d1652c..2abde7c0bec5 100644
--- a/arch/arm64/boot/dts/freescale/fsl-s32g274a-rdb2.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-s32g274a-rdb2.dts
@@ -2,7 +2,7 @@
 /*
  * Device Tree file for S32G274ARDB Board Rev C.
  *
- * Copyright 2020 NXP
+ * Copyright 2020-2021 NXP
  *
  */
 
@@ -14,6 +14,14 @@
 				 "arm,vexpress,v2p-aarch64", "arm,vexpress";
 };
 
+&pfe_reserved_slave {
+	status = "disabled";
+};
+
+&pfe_slave {
+	status = "disabled";
+};
+
 &gmac0 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl0_gmac0_rgmii_a>;
@@ -24,7 +32,7 @@
 };
 
 &pfe1_if {
-	/* PFE0 can be used as SGMII(disabled by default) */
+	/* PFE1 can be used as SGMII(disabled by default) */
 	fsl,pfeng-hif-channel = <1>;
 	phy-mode = "sgmii";
 	status = "disabled";
diff --git a/arch/arm64/boot/dts/freescale/fsl-s32g274a.dtsi b/arch/arm64/boot/dts/freescale/fsl-s32g274a.dtsi
index a5ed62c55a1a..52ec95be0778 100644
--- a/arch/arm64/boot/dts/freescale/fsl-s32g274a.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-s32g274a.dtsi
@@ -43,6 +43,12 @@
 			no-map;
 			status = "okay";
 		};
+		pfe_reserved_slave: pfebufs@83c00000 {
+			compatible = "shared-dma-pool";
+			reg = <0 0x83c00000 0 0x400000>;
+			no-map;
+			status = "disabled";
+		};
 	};
 
 	clks: clks@40038000 {
@@ -104,6 +110,7 @@
 			fsl,pfeng-if-name = "pfe0";
 			fsl,pfeng-emac-id = <0>; /* PFE_EMAC_0 */
 			fsl,pfeng-hif-channel = <0>; /* PFE_HIF_0 */
+			fsl,pfeng-ihc; /* IHC over PFE_HIF_0 */
 			pfe0_mdio: mdio@0 {
 				compatible = "fsl,pfeng-mdio";
 				#address-cells = <1>;
@@ -161,6 +168,40 @@
 		};
 	};
 
+	/* Example of PFE slave driver entry. Disabled by default */
+	pfe_slave: pfe_slave@46000000 {
+		compatible = "fsl,s32g274a-pfeng-slave";
+		reg = <0x0 0x46000000 0x0 0x1000000>;	 /* PFE controller */
+		#address-cells = <0x01>;
+		#size-cells = <0x00>;
+		interrupt-parent = <&gic>;
+		interrupts =  <0 190 1>, /* hif0 */
+			      <0 191 1>, /* hif1 */
+			      <0 192 1>, /* hif2 */
+			      <0 193 1>, /* hif3 */
+			      <0 194 1>, /* bmu */
+			      <0 195 1>; /* nocpy */
+		interrupt-names = "hif0", "hif1", "hif2", "hif3",
+				  "bmu", "nocpy", "upegpt", "safety";
+		memory-region = <&pfe_reserved_slave>;
+		fsl,pfeng-master-hif-channel = <0>;
+		status = "disabled";
+
+		/* EMAC 0 */
+		pfe0sl_if: ethernet@0 {
+			compatible = "fsl,pfeng-logif";
+			#address-cells = <1>;
+			#size-cells = <0>;
+			status = "okay";
+			reg = <0>;
+			local-mac-address = [ 00 04 9F BE EF 11 ];
+			fsl,pfeng-if-name = "pfe0sl";
+			fsl,pfeng-hif-channel = <3>; /* PFE_HIF_3 */
+			fsl,pfeng-emac-id = <0>; /* PFE_EMAC_0 */
+			fsl,pfeng-ihc;
+		};
+	};
+
 	siul2_0 {
 		compatible = "simple-mfd";
 		#address-cells = <2>;
-- 
2.17.1

